### ISSUE TITLE
kontena-cli tries to interpolate docker-compose variables as stack variables

### ISSUE THREAD

**Issue by superzadeh** [kontena/kontena#2943]
## Steps to reproduce

Create two files in a folder:

```yml
# kontena.yml
stack: demo
version: 1.0.0

services:
  node:
    extends:
      file: docker-compose.yml
      service: node

# docker-compose.yml
version: '2.1'
services:
  node:
    image: node:alpine
    environment:
      - ENV_VAR=$ENV_VAR
```

then run:

```sh
$ ENV_VAR=whatever
$ kontena stack validate
 [error] RuntimeError : Undeclared variable 'ENV_VAR' in c:/repos/kontenabug/docker-compose.yml:5 --       - ENV_VAR=$ENV_VAR
         See C:/Users/charl/.kontena/kontena.log or run the command again with environment DEBUG=true set to see the full exception
```

Expected result:
- the stack should be validated

Actual result:
- the stack is not validated. The kontena-cli throws an error about `ENV_VAR` being undeclared. This should not be the case, since `ENV_VAR` is not a kontena variable, but a "regular" `docker-compose` variable.

**Note**: Adding a `kontena stack` variable definition in the `docker-compose.yml` file allows to validate the stack. This create a non-valid `docker-compose` file, but it makes me think that the `docker-compose` is treated as a `kontena stack` file because of the `extend`.

```diff
# docker-compose.yml
version: '2.1'

+variables:
+  myvar:
+    type: string
+    default: 'test'
+    from:
+      env: ENV_VAR
+    to:
+      env: ENV_VAR

services:
  node:
    image: node:alpine
    environment:
      - ENV_VAR=$ENV_VAR
```

**Comment by SpComb** [comment]
This looks like a regression from #2707 at [`Kontena::Cli::Stacks::YAML::Reader#_external_file`](https://github.com/kontena/kontena/commit/386dd763a6c242057410abbedbbcfe665f872165#diff-6ed1773919188b95f7c93b6a87ef648fL300). It used to pass the extending stack `variables` to the `Reader` for the new stack, but it doesn't do that anymore.

**Comment by kke** [comment]
Should we implement the `variables` thing from `depends` to `extends`? Then it would become:

```yaml
stack: demo
version: 1.0.0
variables:
  env_var:
    type: string
    from:
      prompt: Enter env var

services:
  node:
    extends:
      file: docker-compose.yml
      service: node
      variables:
        ENV_VAR: ${env_var}
```

or do we want to allow undeclared env-variables in compose?

**Comment by jakolehm** [comment]
IMHO extends should behave like it's part of the stack yml. No need to re-declare variables.

**Comment by kke** [comment]
So you should do:

```yaml
stack: demo
version: 1.0.0
variables:
  ENV_VAR:
    type: string
    from:
      prompt: Enter env var
```

if your compose yaml has `$ENV_VAR`?

**Comment by SpComb** [comment]
I think the specific example here would also fail on Kontena 1.3 without any `variables: ...` section in the top-level `kontena.yml`?

The interpolation in the extended `docker-compose.yaml` is fixed in #2950, if you explicitly define the variable in the extending `kontena.yml`. I think that's enough to fix the 1.4 regression?
